salt/orch/prepare-product-migration.sls

```diff
@@ -0,0 +1,104 @@
+{#- Get a list of nodes seem to be down or unresponsive #}
+{#- This sends a "are you still there?" message to all #}
+{#- the nodes and wait for a response, so it takes some time. #}
+
+{%- set default_batch = salt['pillar.get']('default_batch', 5) %}
+{#- tx_update_migration_available needs to be last, as it is still used as a selector til reboot #}
+{%- set migration_grains = ['tx_update_migration_notes', 'tx_update_migration_newversion', 'migration_in_progress', 'tx_update_migration_available'] %}
+
+{%- set nodes_down = salt.saltutil.runner('manage.down') %}
+{%- if nodes_down|length >= 1 %}
+# {{ nodes_down|join(',') }} seem to be down: skipping
+  {%- do salt.caasp_log.debug('CaaS: nodes "%s" seem to be down: ignored', nodes_down|join(',')) %}
+  {%- set is_responsive_node_tgt = 'not L@' + nodes_down|join(',') %}
+{%- else %}
+# all nodes seem to be up
+  {%- do salt.caasp_log.debug('CaaS: all nodes seem to be up') %}
+  {#- we cannot leave this empty (it would produce many " and <empty>" targets) #}
+  {%- set is_responsive_node_tgt = '*' %}
+{%- endif %}
+
+{%- set is_migrateable = is_responsive_node_tgt + ' and G@tx_update_migration_available:true' %}
+
+{%- set is_admin = is_migrateable + ' and G@roles:admin' %}
```

Comment:
Not sure about this. Imagine a node with id `XXXX` is down, so:

```Python
is_responsive_node_tgt = 'not L@' + nodes_down|join(',') = 'not L@XXXX'
is_migrateable = is_responsive_node_tgt + ' and G@tx_update_migration_available:true' =  \
  'not L@XXXX and G@tx_update_migration_available:true'
is_admin = is_migrateable + ' and G@roles:admin' = \
  'not L@XXXX and G@tx_update_migration_available:true and G@roles:admin'
```

So, at the end, the only important part is `G@tx_update_migration_available:true and G@roles:admin`, right? In conclusion:

* I would use `set is_admin = 'G@tx_update_migration_available:true and G@roles:admin'`
* I would fail with an exception if `ca in nodes_down` (because I guess we should not consider that case).

And, as a suggestion, I would use `is_migrateable_admin`, so it clearly states that this target is about a migrateable admin and not just the `admin`.

Reply:
> So, at the end, the only important part is `G@tx_update_migration_available:true and G@roles:admin`, right?

i think so, because we're only handling the admin in this variable, and if it would be down we wouldn't even be able to talk to the salt master IIUC